ngraph/core/include/ngraph/type/element_type.hpp

```diff
@@ -94,8 +95,10 @@ namespace ngraph
             bool operator==(const Type& other) const;
             bool operator!=(const Type& other) const { return !(*this == other); }
             bool operator<(const Type& other) const;
+            bool operator==(const Type_t& other) const;
```

Comment:
I'm not sure if  all of this compare function are not required when `Type` is a "strong value" and has conversion operator to `Type_t`.